src/net_processing.cpp

```diff
@@ -581,18 +588,59 @@ class PeerManagerImpl final : public PeerManager
 
     void ProcessOrphanTx(std::set<uint256>& orphan_work_set) EXCLUSIVE_LOCKS_REQUIRED(cs_main, g_cs_orphans)
         EXCLUSIVE_LOCKS_REQUIRED(!m_peer_mutex);
-    /** Process a single headers message from a peer. */
+    /** Process a single headers message from a peer.
+     *
+     * @param[in]   pfrom     CNode of the peer
+     * @param[in]   peer      The peer sending us the headers
+     * @param[in]   headers   The headers received. Note that this may be modified within ProcessHeadersMessage.
```

Comment:
Hmm, interesting point. I think the intention is not to think of the headers in the vector after calling ProcessHeaderMessage as the output of the function, but rather just that the caller shouldn't assume anything about what is in the vector.  So I think it's better to not have the docs say "[in,out]", given that understanding?

Reply:
The bigger question is if this should be a single param or two (and if so, if that's for a followup):
https://github.com/bitcoin/bitcoin/pull/25717#discussion_r954277672

Whether `headers` is treated as an output or just as an input-that-might-get-modified depends on what the callers do with it.

Reply:
Why modify it if you don't want the caller to use it afterwards?

Is it only modified by the `headers.swap(result.pow_validated_headers)` in `IsContinuationOfLowWorkHeadersSync()` and all callers ignore the result? If yes, then that `swap()` can be removed and the parameters made `const`?

Reply:
@vasild For `IsContinuationOfLowWorkHeadersSync`, the comment currently says:
```
     *  @param[in,out] headers                      The headers to be processed.
     *  @return     True if the passed in headers were successfully processed
     *              as the continuation of a low-work headers sync in progress;
     *              false otherwise.
     *              If false, the passed in headers will be returned back to
     *              the caller.
     *              If true, the returned headers may be empty, indicating
     *              there is no more work for the caller to do; or the headers
     *              may be populated with entries that have passed anti-DoS
     *              checks (and therefore may be validated for block index
     *              acceptance by the caller).
```

For `ProcessHeadersMessage`, the comment reads:

```
     * @param[in]   headers   The headers received. Note that this may be modified within ProcessHeadersMessage.
```

Can you clarify what documentation you think is missing?

Reply:
Done.

Reply:
@sdaftuar, I was just briefly looking around and I missed the explanations in `IsContinuationOfLowWorkHeadersSync()` `@return`. The comment in `ProcessHeadersMessage()` is too scarce - "may be modified", but in which way?

Anyway, too minor and I guess the @sipa's suggestion above has improved it enough.